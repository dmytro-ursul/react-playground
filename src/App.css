/* Global App Styles */
.App {
  min-height: 100vh;
  background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
}

/* Login Form Styling */
.login-container {
  display: flex;
  justify-content: center;
  align-items: center;
  min-height: 100vh;
  padding: 2rem;
}

.login-form {
  background: rgba(255, 255, 255, 0.95);
  padding: 3rem;
  border-radius: 20px;
  box-shadow: 0 20px 60px rgba(0, 0, 0, 0.1);
  backdrop-filter: blur(10px);
  border: 1px solid rgba(255, 255, 255, 0.2);
  width: 100%;
  max-width: 400px;
}

.login-form h2 {
  text-align: center;
  margin-bottom: 2rem;
  color: #333;
  font-size: 2rem;
  font-weight: 600;
}

.login-form .form-group {
  margin-bottom: 1.5rem;
}

.login-form label {
  display: block;
  margin-bottom: 0.5rem;
  color: #555;
  font-weight: 500;
}

.login-form input[type="text"],
.login-form input[type="password"] {
  width: 100%;
  padding: 1rem;
  border: 2px solid rgba(102, 126, 234, 0.2);
  border-radius: 12px;
  font-size: 1rem;
  background: rgba(255, 255, 255, 0.9);
  transition: all 0.3s ease;
  outline: none;
  box-sizing: border-box;
}

.login-form input[type="text"]:focus,
.login-form input[type="password"]:focus {
  border-color: #667eea;
  background: white;
  box-shadow: 0 0 0 3px rgba(102, 126, 234, 0.1);
}

.login-form button {
  width: 100%;
  padding: 1rem;
  background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
  color: white;
  border: none;
  border-radius: 12px;
  font-size: 1.1rem;
  font-weight: 600;
  cursor: pointer;
  transition: all 0.3s ease;
  text-transform: uppercase;
  letter-spacing: 0.5px;
}

.login-form button:hover {
  transform: translateY(-2px);
  box-shadow: 0 8px 25px rgba(102, 126, 234, 0.4);
}

.error-message {
  background: rgba(255, 107, 107, 0.1);
  color: #e74c3c;
  padding: 1rem;
  border-radius: 8px;
  margin-bottom: 1rem;
  border: 1px solid rgba(255, 107, 107, 0.2);
  text-align: center;
}
