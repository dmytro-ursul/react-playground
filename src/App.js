import React, {Component} from 'react';
import NewProjectForm from './components/todoList/NewProjectForm';
import TodoList from './components/todoList/TodoList';
// import { createStore } from 'redux'


class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      projects: []
    };
  };

  addNewProject = (name) => {
    const project = { name, items: [] }
    const projects = [...this.state.projects, project]
    this.setState({projects})
  };

  render() {
    return (
      <div className="App">
        <NewProjectForm></NewProjectForm>
	if (projects.length == 0){
	  <p>nothing to show</p>
	} else {
	  <TodoList projects={this.state.projects}></TodoList>
	}
      </div>
    );
  };
}

export default App;
