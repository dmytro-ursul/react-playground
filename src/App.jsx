import TodoList from 'components/todoList/TodoList';
import Login from 'components/Login';

import {
  Routes,
  Route
} from "react-router-dom";

function App(props) {
  return (
    <div>
      <Routes>
        <Route path="/" element={<TodoList/>} />
        <Route path="/login" element={<Login/>} />
      </Routes>
    </div>
  );
}

export default App;
