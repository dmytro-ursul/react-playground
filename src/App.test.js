import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import reducer from './components/todoList/reducers'

let initialState = {
  projects: [ {id: 12345, name: 'initial', items: []} ]
}

const store = createStore(reducer, initialState);


it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App/>
    </Provider>,
    div
  );
});
