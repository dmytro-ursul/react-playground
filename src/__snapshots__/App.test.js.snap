// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="App"
>
  <form
    id="new-project-form"
    onSubmit={[Function]}
  >
    <input
      autoFocus={true}
      onChange={[Function]}
      value=""
    />
    <input
      onClick={[Function]}
      type="button"
      value="add project"
    />
  </form>
  <div
    id="project-list"
  >
    <div
      className="project"
    >
      <div
        className="project-header"
      >
        <p
          className="project-name"
          onClick={[Function]}
        >
          initial
          <span
            aria-hidden={true}
            className="remove-item"
            onClick={[Function]}
            role="button"
          >
            x
          </span>
        </p>
      </div>
      <form
        className="task-form"
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Please enter task name"
          type="text"
          value=""
        />
        <input
          onClick={[Function]}
          type="button"
          value="add task"
        />
      </form>
      <div
        className="task-list"
      />
    </div>
  </div>
</div>
`;
